reference-implementation/lib/transform-stream.js

```diff
@@ -363,14 +386,18 @@ module.exports = class TransformStream {
 
     this._readable = new ReadableStream(source, transformer.readableStrategy);
 
-    const sink = new TransformStreamSink(this, startPromise);
+    const sink = new TransformStreamSink(this, startPromise.then(() => {
+      return TransformStreamBackpressureGonePromise(this);
+    }));
```

Comment:
I felt that it's more readable to me to place logic exchanging signal between sink and source or doing something based on stuff stored inside TransformStream. TransformStream is already controlling the completion of sink.start by passing startPromise. I was viewing this as an additional delaying logic from TransformStream. And, this signal is based on observation on the source. That's my thought behind this suggestion.

But your points also makes sense. So, I'll go with your original structure.